src/ethereum_test_tools/tests/test_filling/test_expect.py

```diff
@@ -0,0 +1,234 @@
+"""
+Test suite for `ethereum_test_tools.filling` fixture post (expect) section.
+"""
+from typing import Any, Mapping, Optional
+
+import pytest
+
+from ethereum_test_forks import Shanghai
+from evm_transition_tool import FixtureFormats, GethTransitionTool
+
+from ...common import Account, Environment, TestAddress, Transaction, to_address
+from ...common.types import Storage
+from ...spec import StateTest
+
+# from ...spec.blockchain.types import FixtureCommon as BlockchainFixtureCommon
+
+ADDRESS_UNDER_TEST = to_address(0x01)
+pre = {
+    TestAddress: Account(balance=1000000000000000000000),
+}
+post = {}
+
+
+def run_post_state_mismatch_test(
+    pre: Mapping[Any, Any], post: Mapping[Any, Any], exception: Any
+) -> Optional[pytest.ExceptionInfo]:
+    """
+    Performs test filling and post state verification.
+    """
+    fork = Shanghai
+    t8n = GethTransitionTool()
+    fixture_format = FixtureFormats.BLOCKCHAIN_TEST
+    state_test = StateTest(
+        env=Environment(),
+        pre=pre,
+        post=post,
+        tx=Transaction(),
+        tag="post_value_mismatch",
+        fixture_format=fixture_format,
+    )
+
+    if exception:
+        e_info: pytest.ExceptionInfo
+        with pytest.raises(exception) as e_info:
+            state_test.generate(t8n=t8n, fork=fork)
+        return e_info
+    else:
+        return None
```

Comment:
Imo, we still have to call `state_test.generate()` here, in order to check that we **don't get an exception**.
```suggestion
    else:
        state_test.generate(t8n=t8n, fork=fork)
        return None
```